image/image_test.go

```diff
@@ -249,3 +254,68 @@ func TestCheckFileInfo(t *testing.T) {
 		t.Error("sha512 should not be matched")
 	}
 }
+
+func TestUncompressedContentSize(t *testing.T) {
+	contentSize := int64(2 * 1024)
+
+	dir, err := createTestDir("testDir1", contentSize)
+	if err != nil {
+		t.Fatalf("Can't create test dir: %v", err)
+	}
+
+	defer os.Remove(dir)
+
+	tarFile1 := path.Join(workDir, "archive1.tar")
+
+	if output, err := exec.Command("tar", "-C", dir, "-cf", tarFile1, "./").CombinedOutput(); err != nil {
+		t.Fatalf("Can't create tar archive: %v", fmt.Sprintf("error: %v, code: %v", string(output), err))
+	}
+	defer os.Remove(tarFile1)
+
+	size, err := image.GetUncompressedTarContentSize(tarFile1)
+	if err != nil {
+		t.Fatalf("Can't get tar content size: %v", err)
+	}
+
+	if contentSize+dirSymlinkSize+dirSymlinkSize != size {
+		t.Errorf("Unexpected tar content size: %v", size)
+	}
+
+	tarFile2 := path.Join(workDir, "archive2.tar.gz")
+
+	if output, err := exec.Command("tar", "-C", dir, "-czf", tarFile2, "./").CombinedOutput(); err != nil {
+		t.Fatalf("Can't create tar archive: %v", fmt.Sprintf("error: %v, code: %v", string(output), err))
+	}
+	defer os.Remove(tarFile2)
+
+	if size, err = image.GetUncompressedTarContentSize(tarFile2); err != nil {
+		t.Fatalf("Can't get tar content size: %v", err)
+	}
+
+	if contentSize+dirSymlinkSize+dirSymlinkSize != size {
+		t.Errorf("Unexpected tar content size: %v", size)
+	}
+}
+
+func createTestDir(dirName string, sizeContent int64) (string, error) {
+	tmpFolder := path.Join(workDir, dirName)
```

Comment:
Here and following use `filepath.Join` instead of `path.Join`: https://stackoverflow.com/questions/39181790/whats-the-difference-between-path-path-filepath-packages-in-go

Reply:
Done